src/mame/sinclair/spectrum.cpp

```diff
@@ -807,7 +810,9 @@ void spectrum_state::spectrum_common(machine_config &config)
 
 	/* sound hardware */
 	SPEAKER(config, "mono").front_center();
-	SPEAKER_SOUND(config, "speaker").add_route(ALL_OUTPUTS, "mono", 0.50);
+	SPEAKER_SOUND(config, m_speaker).add_route(ALL_OUTPUTS, "mono", 0.50);
+	static const double speaker_levels[4] = { 0.0, 1.0, -1.0, 0.0 };
```

Comment:
pushed { 0.0, 0.66, 0.33, 1.0 } as MIC must be louder. Unfortunately I have no expertise to validate exact numbers.

Reply:
Are you sure MIC is louder? Then most games would be using MIC as speaker out to get a louder volume.

Reply:
and according to zxdesign.info:
> The Speaker, Ear socket and Mic socket are all connected to the same ULA pin. Setting bit 4 of port 0xFE high produces a larger voltage at the pin than setting bit 3. For this reason bit 4 is used for sound generation as it will produce the louder sound, and hence it has the 'speaker' label, but you could use bit 3.

Reply:
I’m going to merge this, but the wording, “The Speaker, Ear socket and Mic socket are all connected to the same ULA pin,” makes me think that the mix of both bits should go to the cassette as well.  Should it clip?  It’s probably using AC coupling that we aren’t emulating, too.

Reply:
This one for speaker output. cassette is different story and I can't imagine why somebody is using M.A.M.E. to emulate cassette recording. Maybe some day we'll see report in MT.

Reply:
Saving BASIC programs?  Transferring data to a real machine?  I’ve used MAME to produce a WAV file for an MSX tape program, and then sent it out to my real PX7 using Quicktime Player.

Reply:
> I’m going to merge this, but the wording, “The Speaker, Ear socket and Mic socket are all connected to the same ULA pin,” makes me think that the mix of both bits should go to the cassette as well. Should it clip? It’s probably using AC coupling that we aren’t emulating, too.

I'm bad at analogue tech, perhaps someone skilled may be interested to check ZX-Spectrum48 schematics ? https://spectrumforeveryone.com/wp-content/uploads/2017/08/ZXSpectrumIssue6a-Schematics.gif

at the side note: not all Spectrums have it like that, 128K and later models ULA have 2 pins - EAR and MIC

Reply:
So there’s some low-pass filtering for SPEAKER, EAR and MIC from R90 and C35.  EAR and MIC are both AC-coupled.  SPEAKER isn’t – it’s using a DC-coupled emitter follower.